test/jdk/java/security/Signature/ByteBuffers.java

```diff
@@ -43,11 +45,12 @@ public static void main(String[] args) throws Exception {
         byte[] t = new byte[n];
         random.nextBytes(t);
 
-        KeyPairGenerator kpg = KeyPairGenerator.getInstance("DSA", p);
-        kpg.initialize(512);
+        String kpgAlgorithm = "DSA";
+        KeyPairGenerator kpg = KeyPairGenerator.getInstance(kpgAlgorithm, p);
+        kpg.initialize(SecurityUtils.getTestKeySize(kpgAlgorithm));
         KeyPair kp = kpg.generateKeyPair();
 
-        Signature sig = Signature.getInstance("DSA", p);
+        Signature sig = Signature.getInstance("SHA224withDSA", p);
```

Comment:
Why is SHA224 used instead of SHA256 here and on other tests in this PR?

Reply:
I will update to use SHA256 instead, thanks